core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java

```diff
@@ -2475,4 +2475,21 @@ public int getStaleJobThresholdInDays() {
     public String getIntersectFilterOrSeparator() {
         return getOptional("kylin.query.intersect.separator", "|");
     }
+
+    public String getCanarySinkClass() {
+        return getOptional("kylin.canary.sink.class", "org.apache.kylin.tool.KylinCanary$StdOutSink");
+    }
+
+    public boolean isCanaryDaemon() {
+        return Boolean.parseBoolean(getOptional("kylin.canary.daemon", FALSE));
+    }
+
+    public long getCanaryInterval() {
+        return Long.parseLong(getOptional("kylin.canary.interval", "6000"));
+    }
+
+    public String getCanaryToken() {
+        return getOptional("kylin.canary.token", "");
```

Comment:
Keep the basic auth in configuration file will increase the credential exposure risk. Is there a better way to do the authentication?